src/query/RelayQuery.js

```diff
@@ -211,14 +211,20 @@ class RelayQueryNode {
           if (concreteChild == null) {
             return;
           }
-          var node = createNode(
-            concreteChild,
-            this.__route__,
-            this.__variables__
+
+          let nodes = [].concat(
+            createNode(
+              concreteChild,
+              this.__route__,
+              this.__variables__
+            )
           );
-          if (node && node.isIncluded()) {
-            nextChildren.push(node);
-          }
+
+          nodes.forEach(node => {
+            if (node && node.isIncluded()) {
+              nextChildren.push(node);
+            }
+          });
```

Comment:
I suspect that `[].concat(...)` will be slower than a manual check, especially since most cases will return a single value not an array. Let's explicitly check for array/scalar similar to what you did below.